src/seml/utils/__init__.py

```diff
@@ -397,14 +423,14 @@ def format(self, record):
 
 
 class Hashabledict(dict):
-    def __hash__(self):
+    def __hash__(self):  # type: ignore - I don't think we can satisfy this. This is indeed a hack.
```

Comment:
I'm pretty sure you can't, `Hashabledict` is, strictly speaking, not a proper subtype of `dict`, as it does not implement its interface for `hash`.